app/services/pdf_parser.rb

```diff
@@ -0,0 +1,25 @@
+require 'pdf-reader'
+
+class PdfParser
+  def initialize(file, user)
+    @file = file
+    @user = user
+  end
+
+  def load_residents
+    reader = PDF::Reader.new(@file[:tempfile].path)
+    skip_words = ['information', 'page', 'Unknown','disciple', 'Neighborhood', 'Powered', 'credit card']
+
+    reader.pages.each do |page|
+      page.text.each_line do |line|
+        # Remove blank lines or lines containing skip words
+        next if line.strip.empty? || skip_words.any? { |word| line.include?(word) }
```

Comment:
This is SO much better! Great progress!

Reply:
Here's a more complete example of what I suggested in my text.
Pull the logic for parsing the PDF into a method and just have that return an array of strings. The method just reads the @file and returns an array of all the lines, then you can filter it and create Residents as needed.
In the test, you can do:
```ruby
allow(subject).to receive(:parse_pdf).and return(['foobar', 'Tommy Jones', 'information', 'Elon Musk', 'etc.'])
expect { subject.load_residents }.to change(Resident, :count).by(2)
```